chainer/functions/array/copy.py

```diff
@@ -36,24 +41,38 @@ def copy(x, dst):
     Args:
         x (:class:`~chainer.Variable` or :ref:`ndarray`):
             Variable to be copied.
-        dst (int): Target device specifier.
+        dst: Target device specifier.
 
     Returns:
         ~chainer.Variable: Output variable.
 
     .. admonition:: Example
 
         >>> import chainer.backends.cuda as cuda
-        >>> x = np.random.uniform(-1, 1, (5, 10))
-        >>> cuda.get_device_from_array(x).id
-        -1
-        >>> y = F.copy(x, 0) # from host to device0
-        >>> cuda.get_device_from_array(y.array).id
-        0
-        >>> z = F.copy(y, -1) # from device0 to host
-        >>> cuda.get_device_from_array(z.array).id
-        -1
+        >>> x_arr = np.random.uniform(-1, 1, (5, 10))
+        >>> x = chainer.Variable(x_arr)
+        >>> x.device
+        '@numpy'
+        >>> y = F.copy(x, '@cupy:0') # from CPU (NumPy) to GPU 0 (CuPy)
+        >>> y.device
+        '@cupy:0'
+
+    .. note::
+        Copies between non-ChainerX devices and ChainerX devices are not
+        supported.
```

Comment:
Could you explain why it's not supported (maybe not in the documentation but in this PR)?

Reply:
Because it would be extremely difficult to support backprop.

Reply:
I see, it doesn't get along very well with the `forward_chainerx` forward implementation separation and the fallback mechanism I guess.